data-prepper-test-common/src/main/java/com/amazon/dataprepper/test/helper/ReflectivelySetField.java

```diff
@@ -0,0 +1,34 @@
+package com.amazon.dataprepper.test.helper;
+
+import java.lang.reflect.Field;
+
+/**
+ * Utility class to reflectively set the field of a JSON-based configuration class. Useful for testing input validation behavior.
+ */
+public class ReflectivelySetField {
+    private ReflectivelySetField() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    /**
+     * Reflectively set the field of a configuration object.
+     * @param configurationClass The class of the configuration object.
+     * @param configurationObject The configuration object itself (for tests you might need to specify
+     *                            "NameOfTestClass.this.configurationObject"
+     * @param fieldName Field to change
+     * @param value Value that field is set to
+     *
+     * @throws NoSuchFieldException When field does not exist
+     * @throws IllegalAccessException When field cannot be accessed
+     */
+    public static <T> void setField(final Class<T> configurationClass, final Object configurationObject,
```

Comment:
Thanks! It would be good to include a unit test in this project for this method.

It could ensure the value is set. And further ensure that the field is no longer accessible afterward.